libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -327,21 +321,19 @@ void CompilerUtils::memoryCopy()
 
 	m_context.appendInlineAssembly(R"(
 		{
-		// copy 32 bytes at once
-		start32:
-		jumpi(end32, lt(len, 32))
-		mstore(dst, mload(src))
-		dst := add(dst, 32)
-		src := add(src, 32)
-		len := sub(len, 32)
-		jump(start32)
-		end32:
-
-		// copy the remainder (0 < len < 32)
-		let mask := sub(exp(256, sub(32, len)), 1)
-		let srcpart := and(mload(src), not(mask))
-		let dstpart := and(mload(dst), mask)
-		mstore(dst, or(srcpart, dstpart))
+			// copy 32 bytes at once
+			for {} iszero(lt(len, 32)) {
```

Comment:
I think this could made a bit more readable with:
```
for
    {}
    iszero(lt(len, 32))
    {
       dst...
    }
    {
       mstore...
    }
```